staking/programs/integrity-pool/tests/integrity_pool_slash.rs

```diff
@@ -188,3 +208,224 @@ fn test_create_slash_event() {
     assert_eq!(slash_account_2.slash_custody, slash_custody);
     assert_eq!(slash_account_2.publisher, slashed_publisher);
 }
+
+#[test]
+fn test_slash() {
+    let SetupResult {
+        mut svm,
+        payer,
+        pyth_token_mint,
+        publisher_keypair,
+        pool_data_pubkey,
+        reward_program_authority,
+        publisher_index,
+    } = setup(SetupProps {
+        init_config:     true,
+        init_target:     true,
+        init_mint:       true,
+        init_pool_data:  true,
+        init_publishers: true,
+    });
+
+    let slash_custody = create_token_account(&mut svm, &payer, &pyth_token_mint.pubkey()).pubkey();
+
+    let stake_account_positions =
+        create_stake_account(&mut svm, &payer, &pyth_token_mint, true, true);
+
+    // delegate 10 PYTH at epoch N
+    delegate(
+        &mut svm,
+        &payer,
+        publisher_keypair.pubkey(),
+        pool_data_pubkey,
+        stake_account_positions,
+        10 * FRAC_64_MULTIPLIER,
+    );
+
+    advance_n_epochs(&mut svm, &payer, 1);
+
+    let publisher_caps = post_publisher_caps(&mut svm, &payer, publisher_keypair.pubkey(), 50);
+    advance(&mut svm, &payer, publisher_caps, pyth_token_mint.pubkey()).unwrap();
+
+    advance_delegation_record(
+        &mut svm,
+        &payer,
+        publisher_keypair.pubkey(),
+        stake_account_positions,
+        pyth_token_mint.pubkey(),
+        pool_data_pubkey,
+    )
+    .unwrap();
+
+    // undelegate 5 PYTH at epoch N + 1
+    undelegate(
+        &mut svm,
+        &payer,
+        publisher_keypair.pubkey(),
+        pool_data_pubkey,
+        stake_account_positions,
+        0,
+        5 * FRAC_64_MULTIPLIER,
+    )
+    .unwrap();
+
+    advance_n_epochs(&mut svm, &payer, 1);
+
+    let pool_data: PoolData = fetch_account_data_bytemuck(&mut svm, &pool_data_pubkey);
+
+    // at epoch N + 1 before slashing -> total = 10 pyth, delta = -5 pyth
+    assert_eq!(
+        pool_data.del_state[publisher_index],
+        DelegationState {
+            total_delegation: 10 * FRAC_64_MULTIPLIER,
+            delta_delegation: -5 * FRAC_64_MULTIPLIER as i64,
+        }
+    );
+
+    let publisher_caps = post_publisher_caps(&mut svm, &payer, publisher_keypair.pubkey(), 50);
+    advance(&mut svm, &payer, publisher_caps, pyth_token_mint.pubkey()).unwrap();
+
+    advance_delegation_record(
+        &mut svm,
+        &payer,
+        publisher_keypair.pubkey(),
+        stake_account_positions,
+        pyth_token_mint.pubkey(),
+        pool_data_pubkey,
+    )
+    .unwrap();
+
+    // undelegate 5 PYTH at epoch N + 2
```

Comment:
I don't like that here you're unstaking the whole amount, therefore basically it's all in PREUNLOCKING.
Can we change this so that `total_delegation + delta_delegation !=0`

Reply:
done